cpp/src/arrow/c/bridge.cc

```diff
@@ -2041,6 +2041,19 @@ Status ExportStreamNext(const std::shared_ptr<RecordBatchReader>& src, int64_t i
   }
 }
 
+Status ExportDeviceStreamNext(const std::shared_ptr<RecordBatchReader>& src, int64_t,
```

Comment:
If the `int64_t` is not used, just remove it from the signature?

Reply:
It's in the signature so that we can leverage overloading instead of having to add `if constexpr` checks to pass the `int64_t` or not depending on whether we're using a `RecordBatchReader` or ` ChunkedArray`. Removing it from the signature would prevent being able to use the overloading.

Reply:
Sure, added.